src/textual/design.py

```diff
@@ -0,0 +1,239 @@
+from __future__ import annotations
+
+from typing import Iterable
+
+from rich.console import group
+from rich.padding import Padding
+from rich.table import Table
+from rich.text import Text
+
+from .color import Color, WHITE
+
+NUMBER_OF_SHADES = 3
+
+DEFAULT_DARK_BACKGROUND = "#000000"
+DEFAULT_DARK_SURFACE = "#121212"
+DEFAULT_LIGHT_BACKGROUND = "#f5f5f5"
+DEFAULT_LIGHT_SURFACE = "#efefef"
+
+
+class ColorProperty:
+    """Descriptor to parse colors."""
+
+    def __set_name__(self, owner: ColorSystem, name: str) -> None:
+        self._name = f"_{name}"
+
+    def __get__(
+        self, obj: ColorSystem, objtype: type[ColorSystem] | None = None
+    ) -> Color | None:
+        color = getattr(obj, self._name)
+        if color is None:
+            return None
+        else:
+            return Color.parse(color)
+
+    def __set__(self, obj: ColorSystem, value: Color | str | None) -> None:
+        if isinstance(value, Color):
+            setattr(obj, self._name, value.css)
+        else:
+            setattr(obj, self._name, value)
+
+
+class ColorSystem:
+    """Defines a standard set of colors and variations for building a UI.
+
+    Primary is the main theme color
+    Secondary is a second theme color
+
+
+    """
+
+    COLOR_NAMES = [
+        "primary",
+        "secondary",
+        "background",
+        "surface",
+        "warning",
+        "error",
+        "success",
+        "accent1",
+        "accent2",
+    ]
+
+    def __init__(
+        self,
+        primary: str,
+        secondary: str | None = None,
+        warning: str | None = None,
+        error: str | None = None,
+        success: str | None = None,
+        accent1: str | None = None,
+        accent2: str | None = None,
+        background: str | None = None,
+        surface: str | None = None,
+    ):
+        self._primary = primary
+        self._secondary = secondary
+        self._warning = warning
+        self._error = error
+        self._success = success
+        self._accent1 = accent1
+        self._accent2 = accent2
+        self._background = background
+        self._surface = surface
+
+    @property
+    def primary(self) -> Color:
+        """Get the primary color."""
+        return Color.parse(self._primary)
+
+    secondary = ColorProperty()
+    warning = ColorProperty()
+    error = ColorProperty()
+    success = ColorProperty()
+    accent1 = ColorProperty()
+    accent2 = ColorProperty()
+    surface = ColorProperty()
+    background = ColorProperty()
+
+    @property
+    def shades(self) -> Iterable[str]:
+        """The names of the colors and derived shades."""
+        for color in self.COLOR_NAMES:
+            for shade_number in range(-NUMBER_OF_SHADES, NUMBER_OF_SHADES + 1):
+                if shade_number < 0:
+                    yield f"{color}-darken{abs(shade_number)}"
```

Comment:
Take it or leave it, but CSS frameworks usually put a `-` before the number (Bootstrap, Tailwind, Materialize, BassCSS) and I think it reads nicer (e.g. `primary-darken-2`).

Reply:
Seems reasonable to adopt an existing convention.